src/lib.rs

```diff
@@ -275,20 +275,44 @@ impl File {
 
     /// Persist the tempfile at the specified file path.
     ///
-    /// Note: Because of how temporary file is stored, it cannot be persisted across filesystems.
+    /// NOTE: Because of how temporary file is stored, it cannot be persisted across filesystems.
     /// Also neither the file contents nor the containing directory are
     /// synchronized, so the update may not yet have reached the disk when
     /// `persist_at` returns.
     pub fn persist_at<P: AsRef<Path>>(self, path: P) -> Result<std::fs::File, Error> {
         self.inner.persist(path).map_err(Error::TempFilePersistError)
     }
+
+    /// Create a File from the given buffer.
+    /// Could be used in testing scenarios.
+    pub fn create_from_buffer<P: AsRef<Path>>(buf: &[u8], original_file_name: String, temp_dir: Option<P>) -> Result<Self, Error> {
```

Comment:
Let's call this `new` (or `create`), and please see my other comments for the new signature.